src/proto/Ethereum.proto

```diff
@@ -108,4 +108,7 @@ message SigningOutput {
 
     // The payload part, supplied in the input or assembled from input parameters
     bytes data = 5;
+
+    // The pre-sign tx hash, used for signing
+    bytes pre_hash = 6;
```

Comment:
```suggestion
    bytes hash = 6;
```

since it's already signed hash of the transaction right?

Reply:
No, they are different. First there is a hash from tx encoded in one way, this hash is signed, then tx is encoded again (slightly differently, including sig), and this hash is the hash which is the final hash, also used as TXID in explorer.  I call pre_hash to avoid confusion with final TXID hash.

Reply:
What's the purpose of this change?